core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -101,7 +139,19 @@ public RoleEntity getRoleByIdentifier(NameIdentifier identifier) {
         MetalakeMetaService.getInstance().getMetalakeIdByName(identifier.namespace().level(0));
     RolePO rolePO = getRolePOByMetalakeIdAndName(metalakeId, identifier.name());
 
-    return POConverters.fromRolePO(rolePO, identifier.namespace());
+    List<SecurableObjectPO> securableObjectPOs = listSecurableObjectsByRoleId(rolePO.getRoleId());
+    List<SecurableObject> securableObjects = Lists.newArrayList();
+
+    for (SecurableObjectPO securableObjectPO : securableObjectPOs) {
+      String fullName = getSecurableObjectFullName(securableObjectPO);
+      if (fullName != null) {
```

Comment:
You should add log about the null situation.

Reply:
Done.